pocketfft_hdronly.h

```diff
@@ -152,7 +152,8 @@ template<> struct VLEN<double> { static constexpr size_t val=2; };
 // the __MINGW32__ part in the conditional below works around the problem that
 // the standard C++ library on Windows does not provide aligned_alloc() even
 // though the MinGW compiler and MSVC may advertise C++17 compliance.
-#if (__cplusplus >= 201703L) && (!defined(__MINGW32__)) && (!defined(_MSC_VER))
+// aligned_alloc is only supported from MacOS 10.15.
+#if (__cplusplus >= 201703L) && (!defined(__MINGW32__)) && (!defined(_MSC_VER)) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_15
```

Comment:
```suggestion
#if (__cplusplus >= 201703L) && (!defined(__MINGW32__)) && (!defined(_MSC_VER)) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_15)
```

Reply:
Thanks for the PR!

> Do we need to check whether MAC_OS_X_VERSION_MIN_REQUIRED is defined before doing the comparison? My understanding is that it will evaluate to 0 >= 0 if not on MacOS but correct me if I'm wrong.

To be honest, I don't know; but I just ran a small test and it seems your interpretation is right!
I would have naively expected to get a parser error, but I'm certainly not a preprocessor expert.